src/aks-preview/azext_aks_preview/managed_cluster_decorator.py

```diff
@@ -2037,6 +2054,12 @@ def set_up_addon_profiles(self, mc: ManagedCluster) -> ManagedCluster:
             addon_profiles[
                 CONST_GITOPS_ADDON_NAME
             ] = self.build_gitops_addon_profile()
+
+        if self.context.get_enable_cluster_snapshot():
```

Comment:
why "get_enable_cluster_snapshot" is true, then the addonv2 will be enabled for the items in the list? what's the relation here.

Reply:
There is a toggle to control adding the addonv2 config to use addonManagerV2(with this, we can snapshot addon versions), but that takes quite long time to roll out to everywhere, So after syncing with Tingting, she mentioned that we can directly put the addon profile instead of waiting toggle. So to use snapshot feature, we need to have cluster to running addonManagerV2, in other words, put addonprofile for core addons to enable the addonv2.